src/engine/components/ComponentBase.ts

```diff
@@ -84,23 +72,23 @@ export class ComponentBase {
             this.onEnable();
         }
         let hasUpdate = this.onUpdate.toString().replace(/\s+/g, '').length;
-        if (hasUpdate > 10) {
```

Comment:
you can try define a optional method

```ts
class base{
    protected onUpdate?():void
    onStart(){
        // check if update defined in child
        if(this.onUpdate){
            ....
        }
    }
}
```

Reply:
It may solve #70 as well

Reply:
good idea , I find new method change this .